lib/active_merchant/billing/gateways/ixopay.rb

```diff
@@ -251,6 +253,29 @@ def new_transaction_id
         SecureRandom.uuid
       end
 
+      # Ixopay does not pass any parameters for cardholder/merchant initiated.
+      # Ixopay also doesn't support installment transactions, only recurring
+      # ("RECURRING") and unscheduled ("CARDONFILE").
+      #
+      # Furthermore, Ixopay is slightly unusual in its application of stored
+      # credentials in that the gateway does not return a true
+      # network_transaction_id that can be sent on subsequent transactions.
+
+      def add_stored_credentials(xml, options)
+        return unless stored_credential = options[:stored_credential]
+
+        if stored_credential[:initial_transaction]
+          xml.transactionIndicator 'INITIAL'
+          return
+        end
```

Comment:
Just another approach, but what do you think of something like this here?
```
if stored_credential[:initial_transaction]
  xml.transactionIndicator 'INITIAL'
elsif stored_credential[:reason_type] == 'recurring'
  xml.transactionIndicator 'RECURRING'
elsif stored_credential[:reason_type] == 'unscheduled'
  xml.transactionIndicator 'CARDONFILE'
end
```

Reply:
So my issue here (and the reason for the weird implementation) is that we can take in both `initial_transaction` and `reason_type` but that they both write to the same Ixopay field of `transactionIndicator`. I want to be sure that regardless of anything that the customer sends in for the `reason_type` that if `initial_transaction` is sent that that field will overwrite everything to be certain that we send `INITIAL`. Is there a cleaner way of ensuring that behavior?

Reply:
Hmm, I think it'll do that in the code I wrote, if I understand right. I'm down to talk it over right quick if I'm missing something, though!

Reply:
You're completely right! I've simplified this method now and updated [the unit tests](https://github.com/activemerchant/active_merchant/pull/3549/files#diff-332f420dc6b50a19dc1f41508e903506R271) to confirm that this field is getting passed correctly. Thanks for the suggestion!

Reply:
FWIW I like that this makes it a little easier to understand that it's the same field getting set in the XML regardless of what's put into it. :)